openwisp_monitoring/device/admin.py

```diff
@@ -377,6 +377,9 @@ def related_organization(self, obj):
 
     related_organization.short_description = _('organization')
 
+    def has_delete_permission(self, request, obj=None):
+        return super(admin.ModelAdmin, self).has_delete_permission(request, obj)
```

Comment:
Do you think this is a bug in ReadOnlyAdmin? May it make sense to solve this problem at the root?

Reply:
If this is not an intentional feature, then yes.